devito/ir/clusters/analysis.py

```diff
@@ -119,8 +135,11 @@ class Affiness(Detector):
     Detect the AFFINE Dimensions.
     """
 
-    def _callback(self, clusters, d, prefix):
-        scope = self._fetch_scope(clusters)
+    def _callback(self, clusters, dim, prefix):
+        scope = Scope.fetch(flatten(c.exprs for c in clusters))
```

Comment:
you could move `flatten(c.exprs for c in clusters)` inside `fetch` since it seems to be needed every time.